src/lib/desmo-module.ts

```diff
@@ -87,41 +84,48 @@ export class Desmo {
   }
 
   private async fetchAppOrder(appAddress: string): Promise<AppOrder> {
+    if (this.iexec === undefined) {
+      throw new Error('A connection to iExec is required!');
+    }
     const { orders: appOrders } = await this.iexec.orderbook.fetchAppOrderbook(
       appAddress,
     );
 
-    const appOrder = appOrders && appOrders[0] && appOrders[0].order;
-
-    if (!appOrder) {
-      throw Error(`no apporder found for app ${appAddress}`);
+    if (appOrders.length <= 0) {
+      throw new Error(`no apporder found for app ${appAddress}`);
     } else {
-      return appOrder;
+      return appOrders[0].order as AppOrder;
     }
   }
 
   private async fetchWorkerPoolOrder(): Promise<WorkerpoolOrder> {
+    if (this.iexec === undefined) {
+      throw new Error('A connection to iExec is required!');
+    }
     const { orders: workerpoolOrders } =
       await this.iexec.orderbook.fetchWorkerpoolOrderbook({
         category: this.category,
       });
 
-    const workerpoolOrder =
-      workerpoolOrders && workerpoolOrders[0] && workerpoolOrders[0].order;
-
-    if (!workerpoolOrder) {
-      throw Error(`no workerpoolorder found for category ${this.category}`);
+    if (workerpoolOrders.length <= 0) {
+      throw new Error(`no workerpoolorder found for category ${this.category}`);
     } else {
-      return workerpoolOrder;
+      return workerpoolOrders[0].order as WorkerpoolOrder;
     }
   }
 
   private async retrieveTaskID(): Promise<string> {
+    if (this.iexec === undefined) {
+      throw new Error('A connection to iExec is required!');
+    }
     const deal = await this.iexec.deal.show(this.dealId);
-    return (this.taskId = deal.tasks['0']);
+    return deal.tasks['0'];
```

Comment:
`dealt.tasks` is an array? or an object with `'0'` as property?

Reply:
It's a `Record<number, string>`, so it should be `deal.tasks[0]`. Will fix it now.

Reply:
Weird type 😮 .